product-scenarios/4-gateway/4.1-Extend-the-reach-for-existing-or-legacy-applications/4.1.1-Expose-back-ends-exposed-over-HTTP-as-SOAP-service/4.1.1.1-Expose-a-SOAP-service-as-SOAP-service-using-proxy-service/src/test/java/org/wso2/carbon/esb/scenario/test/ExposeSOAPasSOAPTest.java

```diff
@@ -0,0 +1,95 @@
+ */
+package org.wso2.carbon.esb.scenario.test;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.esb.integration.common.utils.clients.SimpleHttpClient;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * This test class test exposing a SOAP service as SOAP service using proxy service
+ */
+public class ExposeSOAPasSOAPTest extends ScenarioTestBase {
+
+    private static final Log log = LogFactory.getLog(ExposeSOAPasSOAPTest.class);
+
+    private static final String cappNameWithVersion = "scenario_4_1-synapse-configCompositeApplication_1.0.0";
+    private String cappName = "scenario_4_1-synapse-configCompositeApplication";
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        deployCarbonApplication(cappNameWithVersion);
+    }
+
+    @Test(description = "4.1.1.1.1 Front the back-end using pass-through proxy template")
+    public void testPassThroughProxyTemplate() throws IOException, XMLStreamException {
+        invokeSOAPProxyAndAssert(getProxyServiceURLHttp("4_1_1_1_Proxy_SimplePassThroughTemplate"));
+    }
+
+    @AfterClass(description = "Server Cleanup", alwaysRun = true)
+    public void cleanup() throws Exception {
+        super.cleanup();
+        //TODO perform capp undeployment: temporarily disabled till test environment stabilization
+        undeployCarbonApplication(cappName);
+    }
+
+    private void invokeSOAPProxyAndAssert(String url) throws IOException, XMLStreamException {
+        SimpleHttpClient httpClient = new SimpleHttpClient();
+
+        Map<String, String> headers = new HashMap<String, String>(1);
+        headers.put("SOAPAction","urn:getQuote");
+
+        String payload = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" " +
+                "xmlns:ser=\"http://services.samples\" xmlns:xsd=\"http://services.samples/xsd\">\n" +
+                "   <soapenv:Header/>\n" +
+                "   <soapenv:Body>\n" +
+                "      <ser:getQuote>\n" +
+                "         <ser:request>\n" +
+                "            <xsd:symbol>WSO2</xsd:symbol>\n" +
+                "         </ser:request>\n" +
+                "      </ser:getQuote>\n" +
+                "   </soapenv:Body>\n" +
+                "</soapenv:Envelope>";
+
+        HttpResponse response =  httpClient.doPost(url, headers, payload, HTTPConstants.MEDIA_TYPE_TEXT_XML);
```

Comment:
Shall we write a common HTTP Client and make posting a SOAP message generic?

Reply:
+1